### PR TITLE
[API] Cascade and batch multiple runs deletion

### PR DESCRIPTION
**Pull request by alonmr**
Delete run now deletes all of the runtime resources (pods, crds, logs...)
That means that deleting multiple runs should take deletion of the resources into consideration which makes it a heavy operation.
This PR adds async batch runs deletion.

https://jira.iguazeng.com/browse/ML-5307

### CODE REVIEW

File: server/api/crud/runs.py [hunk 1]
```diff
@@ -209,19 +212,76 @@ async def delete_run(
...
+            ]:
+                tasks.append(
+                    server.api.db.session.run_function_with_new_db_session(
+                        self.delete_run,
+                        run.uid,
+                        run.iteration,
+                        run.project,
+                    )
+                )
+            results = await asyncio.gather(*tasks, return_exceptions=True)
```
Comment (TomerShor) [inline server/api/crud/runs.py:267]: Although the tasks run concurrently, the deletions in the DB are still done one by one. 
Perhaps we can still use `db.del_runs` that does the deletion for all runs at once?
We will need to know the UIDs of the runs that succeeded and failed the `delete_run` task.

File: server/api/crud/runs.py [hunk 2]
```diff
@@ -209,19 +212,76 @@ async def delete_run(
...
+                    )
+
+            runs_list = runs_list[
+                mlrun.config.config.crud.runs.batch_delete_runs_chunk_size :
+            ]
+
+        if failed_deletions:
+            raise mlrun.errors.MLRunRuntimeError(
+                f"Failed to delete {failed_deletions} run(s). Error: {mlrun.errors.err_to_str(last_exception)}"
+            ) from last_exception
```
Comment (TomerShor) [inline server/api/crud/runs.py:284]: Showing only the last exception might miss some other errors.

**Review by TomerShor — COMMENTED** [review]
Looking good overall! some minor comments

File: server/api/crud/runs.py (hunk 2 above)
Comment (alonmr) [reply server/api/crud/runs.py:284]: I didn't want to make the error endless in case many runs failed to delete. Either way the errors are logged above.

**Review by TomerShor — APPROVED** [review]
LGTM

**Review by theSaarco — APPROVED** [review]
Looks good. Have some minor suggestions, but generally approved.

**Pull request merged**